src/ModelInfo.php

```diff
@@ -12,6 +14,20 @@
 
 class ModelInfo
 {
+    protected static array $typeMappings = [
```

Comment:
Having this all implemented as `static` seems a bit heavy. Could this be handled another way?

Reply:
I do agree. However, both of the places it is needed are referenced from a static context as well. One solution would be to move the calculations for the properties inside the constructor definition, and that would allow `registerTypeMappings` to become non-static. Moving `$typeMappings` to a non-static context would make extending them more difficult, as they would have to be passed to `forAllModels` and `forModel`, which IMO would clutter up the function calls. 
Another option is to move to a dedicated config where they can be configured. `$directory`,`$basePath`, and `$baseNamespace` could also move there as well to allow for a default configuration of the package.

Reply:
> both of the places it is needed are referenced from a static context as well

Could you clarify this?

Reply:
`ModelInfo::forModel` and `ModelInfo::forAllModels` are both static, and `registerTypeMappings` needs to be called on the database connection for the model before the models column information is fetched from the database in `ModelInfo::forModel`

Reply:
Ok, thanks for explaining, let's leave it like it is for now. Could you also add a test for this?